File: src/PwrDrvr.MicroApps.DeployTool/DeployConfig.cs
using System;
using System.Globalization;
using System.Linq.Expressions;
using System.Text.Json;
using System.IO;

namespace PwrDrvr.MicroApps.DeployTool {
  public class DeployConfig {
    private static readonly string _fileName = "deploy.json";

    internal static DeployConfig Load() {
      if (File.Exists(DeployConfig._fileName)) {
        var config = JsonSerializer.Deserialize<DeployConfig>(File.ReadAllText(DeployConfig._fileName));
        return config;
      }
      return null;
    }

    public string DefaultFile { get; set; }
    private string _appName;
    public string AppName {
      get {
        return _appName;
      }
      set {
        _appName = value.ToLower();
      }
    }
    public string SemVer { get; set; }
    public string StaticAssetsPath { get; set; }
    public string LambdaARN { get; set; }
  }
}

File: src/microapps-publish/src/DeployConfig.ts
import { promises as fs } from 'fs';

export interface IDeployConfig {
  AppName: string;
  SemVer: string;
  DefaultFile: string;
  StaticAssetsPath: string;
  LambdaARN: string;
  AWSAccountID: string;
  AWSRegion: string;
  ServerlessNextRouterPath: string;
}

export default class DeployConfig implements IDeployConfig {
  private static readonly _fileName = 'deploy.json';

  public static async Load(): Promise<DeployConfig> {
    try {
      const stat = await fs.stat(DeployConfig._fileName);
      if (stat.isFile()) {
        const config = JSON.parse(
          await fs.readFile(DeployConfig._fileName, 'utf-8'),
        ) as DeployConfig;
        return config;
      }
    } catch {
      // File did not exist, so stat throws
      return undefined;
    }
    return undefined;
  }

  private _appName: string;
  public get AppName(): string {
    return this._appName;
  }
  public set AppName(value: string) {
    this._appName = value.toLowerCase();
  }
  public SemVer: string;
  public StaticAssetsPath: string;
  public LambdaARN: string;
  public AWSAccountID: string;
  public AWSRegion: string;
  public ServerlessNextRouterPath: string;
  public DefaultFile: string;
}
